### PR TITLE
CIF-1611 - Update Venia project structure based on AEM archetype 24

### PR DESCRIPTION
**Pull request by mhaack**
## Description

Update Venia project structure based on AEM archetype 24. This mainly changes the following areas:
* all packages using correct `packageType`
* filevault package & content validation now enabled
* OSGI configs moved from `ui.apps` to new `ui.config` module
* all dependencies are embedded via `all` sub module/package only
* new `ui.test` sub-module

## Related Issue

CIF-1611

## Motivation and Context

Venia project structure should be aligned with AEM project archetype release. Before it was build using a snapshot version.

## How Has This Been Tested?

Deployed and tested project on AEM Cloud SDK & AEM 6.5

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

**Review by dplaton — APPROVED** [review]

File: ui.apps/src/main/content/jcr_root/apps/venia/components/commerce/miniaccount/.content.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
     jcr:primaryType="cq:Component"
-    jcr:title="Mini Account"
+    jcr:title="Mini Cart"
```
Comment (cjelger) [inline ui.apps/src/main/content/jcr_root/apps/venia/components/commerce/miniaccount/.content.xml:4]: That shouldn't have changed, no?

File: ui.content/src/main/content/jcr_root/content/experience-fragments/venia/.content.xml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
     cq:adobeTargetExportFormat="html"
-    cq:allowedTemplates="[/conf/venia/settings/wcm/templates/xf-web-variation]"
     cq:conf="/conf/venia"
+    cq:allowedTemplates="[/conf/venia/settings/wcm/templates/(?!page-).*]"
```
Comment (cjelger) [inline ui.content/src/main/content/jcr_root/content/experience-fragments/venia/.content.xml:5]: Is this correct? I mean, the only allowed template should be the XF one, no?

File: ui.apps/src/main/content/jcr_root/apps/venia/components/commerce/miniaccount/.content.xml (hunk 1 above)
Comment (mhaack) [reply ui.apps/src/main/content/jcr_root/apps/venia/components/commerce/miniaccount/.content.xml]: Yes I reverted it, this is wrong in the AEM archetype as well, created https://github.com/adobe/aem-project-archetype/pull/525 to fix it there as well.

File: ui.content/src/main/content/jcr_root/content/experience-fragments/venia/.content.xml (hunk 2 above)
Comment (mhaack) [reply ui.content/src/main/content/jcr_root/content/experience-fragments/venia/.content.xml]: Fixed it

**Review by herzog31 — APPROVED** [review]

**Review by cjelger — APPROVED** [review]

**Pull request merged**